gensrc/script/functions.py

```diff
@@ -56,6 +56,8 @@
     [10090, "tan", "DOUBLE", ["DOUBLE"], "MathFunctions::tan"],
     [10100, "atan", "DOUBLE", ["DOUBLE"], "MathFunctions::atan"],
     [10101, "tanh", "DOUBLE", ["DOUBLE"], "MathFunctions::tanh"],
+    [10102, "cosine_similarity", "FLOAT", ["ARRAY_FLOAT", "ARRAY_FLOAT"], "MathFunctions::cosine_similarity<TYPE_FLOAT, false>"],
```

Comment:
FLOAT OR DOUBLE?

Reply:
float only. we can extend to `double` when necessary.  

but for most cases in Vectordb, I guess precision does not matter much.